modules/bullet/bullet_physics_server.cpp

```diff
@@ -601,7 +607,10 @@ void BulletPhysicsServer::body_set_collision_layer(RID p_body, uint32_t p_layer)
 	RigidBodyBullet *body = rigid_body_owner.get(p_body);
 	ERR_FAIL_COND(!body);
 
-	body->set_collision_layer(p_layer);
+	if (body->get_collision_layer() != p_layer) {
+		body->set_collision_layer(p_layer);
+		body->set_activation_state(true);
```

Comment:
Following the above advice, can you please put the `set_activation_state` inside the `set_collision_layer` ?